test/provisioner/provision_azure.go

```diff
@@ -8,37 +8,355 @@ package provisioner
 type AzureCloudProvisioner struct {
 }
 
 func NewAzureCloudProvisioner(properties map[string]string) (CloudProvisioner, error) {
+	if err := initazureProperties(properties); err != nil {
+		return nil, err
+	}
+
 	return &AzureCloudProvisioner{}, nil
 }
 
-func (l *AzureCloudProvisioner) CreateCluster(ctx context.Context, cfg *envconf.Config) error {
-	return nil
+func (p *AzureCloudProvisioner) CreateVPC(ctx context.Context, cfg *envconf.Config) error {
+	log.Trace("CreateVPC()")
+	return createResourceImpl()
 }
 
-func (l *AzureCloudProvisioner) CreateVPC(ctx context.Context, cfg *envconf.Config) error {
-	return nil
+func (p *AzureCloudProvisioner) DeleteVPC(ctx context.Context, cfg *envconf.Config) error {
+	log.Trace("DeleteVPC()")
+	return deleteResourceImpl()
 }
 
-func (l *AzureCloudProvisioner) DeleteCluster(ctx context.Context, cfg *envconf.Config) error {
+func (p *AzureCloudProvisioner) CreateCluster(ctx context.Context, cfg *envconf.Config) error {
+	log.Trace("CreateCluster()")
+	agentPoolProperties := []containerservice.ManagedClusterAgentPoolProfile{
+		{
+			Name:               &AzureProps.NodeName,
+			Count:              to.Int32Ptr(2),
+			VMSize:             &AzureProps.InstanceSize,
+			Mode:               containerservice.AgentPoolModeSystem,
+			OsType:             containerservice.OSType(AzureProps.OsType),
+			EnableNodePublicIP: to.BoolPtr(true),
+			VnetSubnetID:       &AzureProps.SubnetID,
+		},
+	}
+
+	servicePrincipalProfile := &containerservice.ManagedClusterServicePrincipalProfile{
+		ClientID: &AzureProps.ClientID,
+		Secret:   &AzureProps.ClientSecret,
+	}
+
+	ManagedClusterProperties := &containerservice.ManagedClusterProperties{
+		ServicePrincipalProfile: servicePrincipalProfile,
+		DNSPrefix:               to.StringPtr("caa-dns-prefix"),
+		AgentPoolProfiles: &[]containerservice.ManagedClusterAgentPoolProfile{
+			agentPoolProperties[0],
+		},
+	}
+
+	ManagedCluster := &containerservice.ManagedCluster{
+		Location:                 to.StringPtr("eastus"),
+		ManagedClusterProperties: ManagedClusterProperties,
+	}
+
+	// Create the cluster
+	log.Infof("Creating cluster %s.\n", AzureProps.ClusterName)
+	_, err := AzureProps.ManagedaksClient.CreateOrUpdate(context.Background(), AzureProps.ResourceGroupName, AzureProps.ClusterName, *ManagedCluster)
+	if err != nil {
+		return err
+	}
+
+	// check if cluster is ready after creation
+	clusterReady := false
+	waitMinutes := 10
+	log.Infof("Waiting for cluster %s to be available.\n", AzureProps.ClusterName)
+	for i := 0; i <= waitMinutes; i++ {
+		ready, err := isAzureClusterReady(AzureProps.ResourceGroupName, AzureProps.ClusterName)
+		if err != nil {
+			log.Warnf("Err %s happened when retrieve cluster, try again...\n", err)
+			continue
+		}
+		if ready {
+			log.Infof("Cluster %s is available.\n", AzureProps.ClusterName)
+			clusterReady = true
+			break
+		}
+
+		time.Sleep(60 * time.Second)
+	}
```

Comment:
I think I would replace this custom polling mechanism would something like the following provided by the package: "k8s.io/apimachinery/pkg/util/wait"

```go
if err = wait.PollImmediate(retryInterval, timeout, func() (done bool, err error) {
	// gotta try again
	return false, nil

	// error occurred while waiting
	return false, err

	// Successful outcome
	return true, nil
}); err != nil {
}
```